sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharpExplodedGraph.cs

```diff
@@ -277,8 +277,12 @@ protected override void VisitInstruction(ExplodedGraphNode node)
                 case SyntaxKind.RefValueExpression:
 
                 case SyntaxKind.MemberBindingExpression:
+                    newProgramState = newProgramState.PopValue();
+                    newProgramState = newProgramState.PushValue(new SymbolicValue());
+                    break;
 
                 case SyntaxKind.AwaitExpression:
+                    newProgramState = newProgramState.RemoveSymbols(IsFieldSymbol);

```

Comment:
Awaiting a task potentially could change the fields, so to avoid FP we clear the field symbols from the current program state.